### ISSUE TITLE
Logging in with an incorrect password shows an unhelpful error message

### ISSUE THREAD

**Issue by garethbowen** [medic/medic#5486]
**Steps to reproduce**:
1. Go to login page
2. Enter an incorrect user/pass combination
3. Submit

**What should happen**:
- An error message is displayed something along the lines of "incorrect user/pass"

**What actually happens**:
- This error message is shown: "Unexpected error while logging in. Please try again."
- There's an error in the API logs

![Screenshot from 2019-03-12 10-13-09](https://user-images.githubusercontent.com/2064938/54158500-77b7a380-44af-11e9-9dd9-18277a3b66be.png)

Console logs:
```
Error logging in {"error":"Unexpected error logging in"} script.js:24:5
	handleResponse https://gamma.dev.medicmobile.org/login/script.js:24
	onreadystatechange https://gamma.dev.medicmobile.org/login/script.js:9
```

API logs:
```
REQ 4022b704-8cb5-4e3e-8cf3-ed8dfc47deb6 ::ffff:127.0.0.1 - POST /medic/login HTTP/1.1
2019-03-12 09:30:20 ERROR: Error logging in: { StatusCodeError: 401 - {"error":"unauthorized","reason":"Name or password is incorrect."}
    at new StatusCodeError (/home/gareth/src/medic/medic/api/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/home/gareth/src/medic/medic/api/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/home/gareth/src/medic/medic/api/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/home/gareth/src/medic/medic/api/node_modules/request/request.js:185:22)
    at Request.emit (events.js:189:13)
    at Request.<anonymous> (/home/gareth/src/medic/medic/api/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:189:13)
    at IncomingMessage.<anonymous> (/home/gareth/src/medic/medic/api/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:277:13)
    at IncomingMessage.emit (events.js:194:15)
  name: 'StatusCodeError',
  statusCode: 401,
  message:
   '401 - {"error":"unauthorized","reason":"Name or password is incorrect."}',
  error:
   { error: 'unauthorized',
     reason: 'Name or password is incorrect.' },
  options:
   { url: 'http://localhost:5984/_session',
     json: true,
     resolveWithFullResponse: true,
     body: { name: 'admin', password: 'passddd' },
     auth: { user: 'admin', pass: 'passddd' },
     method: 'POST',
     callback:
      { [Function: RP$callback]
        [length]: 3,
        [name]: 'RP$callback',
        [prototype]: RP$callback { [constructor]: [Circular] } },
     transform: undefined,
     simple: true,
     transform2xxOnly: false },
  response:
    ... <many lines>
```

**Environment**:

- Instance: gamma.dev.medicmobile.org
- Browser: Chrome (none others tested)
- Client platform: Ubuntu
- App: webapp/api
- Version: 3.4.x, master

**Comment by garethbowen** [comment]
This is ready for AT in the `5486-handle-non-200-login-responses` branch.

**Comment by ngaruko** [comment]
LGTM
Proper message shows in the UI and no error in logs, just a 401 log
![image](https://user-images.githubusercontent.com/6979995/54174656-8f614d00-44ec-11e9-804f-87443ca58490.png)

**Comment by garethbowen** [comment]
Merged and backported to `3.4.x`
